### ISSUE TITLE
[Bug] Missing JSON attribute

### ISSUE THREAD

**Issue by SebSchu95** [iv-org/invidious#2210]
Title: `Missing JSON attribute: simpleText
  parsing CaptionName at line 1, column 350
  parsing Caption#name at line 1, column 343 (JSON::SerializableError)`
Date: `2021-06-26T00:12:30Z`
Route: `/watch?v=z2ms0tKHZcI`
Version: `2021.06.25-ae1d9e8 @ master`
<details>
<summary>Backtrace</summary>
<p>
   
```
Missing JSON attribute: simpleText
  parsing CaptionName at line 1, column 350
  parsing Caption#name at line 1, column 343 (JSON::SerializableError)
  from /usr/share/crystal/src/json/serialization.cr:159:7 in 'initialize:__pull_for_json_serializable'
  from src/invidious/videos.cr:792:3 in 'new_from_json_pull_parser'
  from src/invidious/videos.cr:792:3 in 'new'
  from /usr/share/crystal/src/json/from_json.cr:13:3 in 'from_json'
  from src/invidious/videos.cr:709:17 in 'captions'
  from src/invidious/routes/watch.cr:150:5 in 'handle'
  from src/invidious.cr:299:1 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'process_request'
  from lib/kemal/src/kemal/route_handler.cr:17:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/websocket_handler.cr:13:14 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/filter_handler.cr:21:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:212:5 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:94:12 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:145:12 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:70:5 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/static_file_handler.cr:189:11 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/exception_handler.cr:8:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/logger.cr:17:35 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/init_handler.cr:12:7 in 'call'
  from /usr/share/crystal/src/http/server/request_processor.cr:51:11 in 'process'
  from /usr/share/crystal/src/http/server.cr:513:5 in 'handle_client'
  from /usr/share/crystal/src/http/server.cr:468:13 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'run'
  from /usr/share/crystal/src/fiber.cr:92:34 in '->'
  from ???
Caused by: Missing JSON attribute: simpleText
  parsing CaptionName at line 1, column 350 (JSON::SerializableError)
  from /usr/share/crystal/src/json/serialization.cr:159:7 in 'initialize:__pull_for_json_serializable'
  from src/invidious/videos.cr:786:3 in 'new_from_json_pull_parser'
  from src/invidious/videos.cr:786:3 in 'new'
  from /usr/share/crystal/src/json/serialization.cr:159:7 in 'initialize:__pull_for_json_serializable'
  from src/invidious/videos.cr:792:3 in 'new_from_json_pull_parser'
  from src/invidious/videos.cr:792:3 in 'new'
  from /usr/share/crystal/src/json/from_json.cr:13:3 in 'from_json'
  from src/invidious/videos.cr:709:17 in 'captions'
  from src/invidious/routes/watch.cr:150:5 in 'handle'
  from src/invidious.cr:299:1 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'process_request'
  from lib/kemal/src/kemal/route_handler.cr:17:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/websocket_handler.cr:13:14 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/filter_handler.cr:21:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:212:5 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:94:12 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:145:12 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/handlers.cr:70:5 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/static_file_handler.cr:189:11 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/exception_handler.cr:8:7 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from src/invidious/helpers/logger.cr:17:35 in 'call'
  from /usr/share/crystal/src/http/server/handler.cr:28:7 in 'call_next'
  from lib/kemal/src/kemal/init_handler.cr:12:7 in 'call'
  from /usr/share/crystal/src/http/server/request_processor.cr:51:11 in 'process'
  from /usr/share/crystal/src/http/server.cr:513:5 in 'handle_client'
  from /usr/share/crystal/src/http/server.cr:468:13 in '->'
  from /usr/share/crystal/src/primitives.cr:255:3 in 'run'
  from /usr/share/crystal/src/fiber.cr:92:34 in '->'
  from ???
```
</p>
</details>

**Comment by SamantazFox** [comment]
On which instance did you encounter this issue?

@syeopite Have you deployed #2189 on your instance ?

**Comment by unixfox** [comment]
He probably tested that on yewtu.be. This is due to the testing of https://github.com/iv-org/invidious/pull/2205 which partially work for some age-restricted videos.

Closing as we already know about this bug.

**Comment by test1101111** [comment]
> He probably tested that on yewtu.be. This is due to the testing of #2205 which partially work for some age-restricted videos.
> 
> Closing as we already know about this bug.

Can't therefore this particular bug page contain just a link to this thread/solution, so that no more ppl will come here and made another reports?

**Comment by unixfox** [comment]
> > He probably tested that on yewtu.be. This is due to the testing of #2205 which partially work for some age-restricted videos.
> > Closing as we already know about this bug.
> 
> Can't therefore this particular bug page contain just a link to this thread/solution, so that no more ppl will come here and made another reports?

It's due to an ongoing test related to #2205. This temporary bug will be fixed once #2205 is merged.
